### PR TITLE
[Remove Stream/Optional usage] LPS-178007 remove_stream_and_optional_in_headless_delivery_api

### PR DESCRIPTION
**Pull request by jeffwu0724**
Hi Tina, Dante,

resent: https://github.com/liferay-core-infra/liferay-portal/pull/3893
This is for ticket https://issues.liferay.com/browse/LPS-178007.
Thanks for checking!

### CODE REVIEW

**Comment by jeffwu0724** [comment]
ci:test:headless

File: modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/ContentFieldUtil.java [hunk 1]
```diff
@@ -115,16 +115,16 @@ public static ContentField toContentField(
...
-							Value::getValues
-						).orElse(
-							Collections.emptyMap()
-						);
+						Value value = ddmFormFieldValue.getValue();
+
+						Map<Locale, String> values = value.getValues();
+
+						if (values == null) {
+							values = Collections.emptyMap();
```
Comment (jeffwu0724) [inline modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/ContentFieldUtil.java:123]: rename valueValues to value, also assign `Collections.emtpyMap()` to values if it is null

File: modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/DDMValueUtil.java [hunk 2]
```diff
@@ -448,23 +447,22 @@ private static LocalizedValue _toLocalizedValue(
...
-				Locale locale = LocaleUtil.fromLanguageId(
-					languageId, true, false);
-
-				if (locale != null) {
-					localizedValue.addString(
-						locale,
-						localizedValueBiFunction.apply(
-							localizedContentFieldValue, locale));
-				}
+		if (localizedContentFieldValues == null) {
```
Comment (jeffwu0724) [inline modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/DDMValueUtil.java:450]: assigin to `Collections.emptyMap()` to make it consistent.

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by jeffwu0724** [comment]
ci:help

**Comment by jeffwu0724** [comment]
ci:test:sf

**Comment by jeffwu0724** [comment]
ci:test:headless

**Comment by liuxingye** [comment]
The failed tests from both [ci:test:relevant](https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2239847268) and [ci:test:headless](https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2239624551) test suits are not related to this pull. Could you help to add _QA Passed_ label? Thanks.

cc @dantewang

**Review by dantewang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**